### PR TITLE
feat: reset pin

### PR DESCRIPTION
**Pull request by Paulmburu**
- implement reset pin on expiry

### CODE REVIEW

File: lib/application/redux/actions/send_otp_action.dart [hunk 1]
```diff
@@ -56,46 +53,40 @@ class SendOTPAction extends ReduxAction<AppState> {
 
   @override
   Future<AppState?> reduce() async {
-    final String phoneNumber =
-        state.clientState!.user!.primaryContact!.value ?? UNKNOWN;
+    final bool isResetPin =
+        StoreProvider.state<AppState>(context)?.onboardingState?.isResetPin ??
```
Comment (johnngugi) [inline lib/application/redux/actions/send_otp_action.dart:57]: Access state directly instead of through storeprovider

**Review by johnngugi — CHANGES_REQUESTED** [review]

File: lib/application/core/services/onboarding_utils.dart [hunk 2]
```diff
@@ -61,6 +61,19 @@ ProcessedResponse processHttpResponse(
       );
     }
 
+    if (body['code'] == 48 && context != null) {
+      StoreProvider.dispatch(
+        context,
+        UpdateOnboardingStateAction(isResetPin: true),
+      );
+
+      Navigator.pushNamed(
```
Comment (johnngugi) [inline lib/application/core/services/onboarding_utils.dart:70]: I think navigation should occur within the action instead of inside this function

**Review by johnngugi — CHANGES_REQUESTED** [review]

File: lib/application/core/services/onboarding_utils.dart [hunk 3]
```diff
@@ -24,13 +21,15 @@ import 'package:myafyahub/domain/core/entities/core/onboarding_path_config.dart'
...
 import 'package:myafyahub/presentation/router/routes.dart';
+import 'package:shared_themes/constants.dart';
 
 /// [processHttpResponse] routine is used to process a network call response, for errors, bad requests, timeouts and correct responses.
 ///
 /// errors are further processed using our own custom error codes to determine if it an error within our error codes
 ProcessedResponse processHttpResponse(
   http.Response response, [
   BuildContext? context,
+  String? phoneNumber,
```
Comment (johnngugi) [inline lib/application/core/services/onboarding_utils.dart:32]: Is the phoneNumber param being used?

**Review by johnngugi — CHANGES_REQUESTED** [review]

File: lib/application/core/services/onboarding_utils.dart (hunk 3 above)
Comment (Paulmburu) [reply lib/application/core/services/onboarding_utils.dart:32]: not really, let me get rid of it

**Review by johnngugi — APPROVED** [review]

**Review by EugeneDevv — APPROVED** [review]

**Pull request merged**